src/Pos/DB/Block.hs

```diff
@@ -247,19 +254,40 @@ runBlockDBRedirect = coerce
 
 -- instance MonadBlockDBGeneric (Block ssc)
 
+type BlockDBGenericRealMonad ssc m =
+    (MonadDBRead m, MonadRealDB m, SscHelpersClass ssc)
+
+dbGetBlockReal :: forall ssc m . BlockDBGenericRealMonad ssc m => HeaderHash -> m (Maybe (Block ssc))
+dbGetBlockReal = getBlock
+
+dbGetUndoReal :: forall ssc m . BlockDBGenericRealMonad ssc m => HeaderHash -> m (Maybe Undo)
+dbGetUndoReal = getUndo
+
+dbGetHeaderReal :: forall ssc m . BlockDBGenericRealMonad ssc m => HeaderHash -> m (Maybe (BlockHeader ssc))
+dbGetHeaderReal = blkGetHeader
+
 instance (MonadDBRead m, MonadRealDB m, t ~ IdentityT, SscHelpersClass ssc) =>
          MonadBlockDBGeneric (BlockHeader ssc) (Block ssc) Undo (Ether.TaggedTrans BlockDBRedirectTag t m) where
-    dbGetBlock  = getBlock
-    dbGetUndo   = getUndo
-    dbGetHeader = blkGetHeader
+    dbGetBlock  = dbGetBlockReal @ssc
+    dbGetUndo   = dbGetUndoReal @ssc
+    dbGetHeader = dbGetHeaderReal @ssc
 
 -- instance MonadBlockDBGeneric (SscBlock ssc)
 
+dbGetBlockReal' :: forall ssc m . BlockDBGenericRealMonad ssc m => HeaderHash -> m (Maybe (SscBlock ssc))
+dbGetBlockReal' = fmap (toSscBlock <$>) . getBlock
+
+dbGetUndoReal' :: forall ssc m . BlockDBGenericRealMonad ssc m => HeaderHash -> m (Maybe ())
+dbGetUndoReal' = fmap (const () <$>) . getUndo
+
+dbGetHeaderReal' :: forall ssc m . BlockDBGenericRealMonad ssc m => HeaderHash -> m (Maybe (Some IsHeader))
```

Comment:
If we add such function for `txp`, will we call them `dbGetBlockReal''`? Hopefully no.